apps/core/src/utils/transaction/getGroupByOwner.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    SuiObjectChangeCreated,
+    SuiObjectChangeMutated,
+    SuiObjectChangeTransferred,
+} from '@mysten/sui.js';
+
+export enum LocationIdType {
+    AddressOwner = 'AddressOwner',
+    ObjectOwner = 'ObjectOwner',
+    Shared = 'Shared',
+    Unknown = 'Unknown',
+}
+
+export function getGroupByOwner(
+    objectSummaryChanges: (
+        | SuiObjectChangeMutated
+        | SuiObjectChangeCreated
+        | SuiObjectChangeTransferred
+    )[]
+) {
+    if (!objectSummaryChanges) {
+        return {};
+    }
+
+    return objectSummaryChanges?.reduce(
+        (mapByOwner: Record<string, any[]>, change: any) => {
```

Comment:
why are we typing `change` and the value in `mapByOwner` to any? I'd recommend keeping the strong typing and narrow down the fields you want from `change` using the `type` field. e.g.:

```
const owner = change.type !== 'transferred' ? change.owner : undefined;
...

if (typeof owner === 'object' && 'AddressOwner' in owner) {
...
}
```